pkg/controller/operators/catalog/operator.go

```diff
@@ -622,6 +628,35 @@ func (o *Operator) syncConnection(logger *logrus.Entry, in *v1alpha1.CatalogSour
 	return
 }
 
+func (o *Operator) waitForBackingPod(logger *logrus.Entry, in *v1alpha1.CatalogSource) (out *v1alpha1.CatalogSource, continueSync bool, syncError error) {
+	out = in.DeepCopy()
+
+	selector := map[string]string{reconciler.CatalogSourceLabelKey: in.GetName()}
+	pods, err := o.lister.CoreV1().PodLister().Pods(in.GetNamespace()).List(labels.SelectorFromSet(selector))
+	if err != nil {
+		continueSync = false
+		syncError = fmt.Errorf("error while examining catalog source pod: %v", err)
+		return
+	}
+
+	if len(pods) > 0 {
+		for _, cond := range pods[0].Status.Conditions {
+			if cond.Type == corev1.ContainersReady && cond.Status == corev1.ConditionTrue {
+				continueSync = true
+				return
+			}
+		}
+	} else {
+		continueSync = false
+		syncError = fmt.Errorf("did not find expected backing pod")
+		return
+	}
+
+	logger.Infof("backing pod not yet ready")
```

Comment:
maybe we could add `"pod status %s", pods[0].Status.Message` to this log message?

Reply:
Maybe a good compromise is to indicate the pod name here? The message seems to be blank in my testing and printing the whole status block is going to be too much.